### ISSUE TITLE
Some external crate type cannot be resolved (chrono::Duration)

### ISSUE THREAD

**Issue by EdeMeijer** [intellij-rust/intellij-rust#2666]
## Environment

* **Intellij-Rust plugin version:** 0.2.0.2100-181
* **IDE name and version:** Intellij IDEA ultimate 2018.1.5
* **Operating system:** Ubuntu 16.04

## Problem description
When using the `chrono::Duration` type from the `chrono` crate, the plugin does not acknowledge it's existence which makes development a bit harder. Other types from the `chrono` crate are resolved without problems.

## Steps to reproduce
- Create new cargo project with only `chrono` as a dependency
- Try to use the `chrono::Duration` type

![duration](https://user-images.githubusercontent.com/5758565/42677853-6084db4c-867d-11e8-9dbe-a73ffd105a31.png)

**Comment by kumbayo** [comment]
I can reproduce the problem with 0.2.0.2100-181 and current master a03c862f52264692d2935369a10f18635679d8ac
It seems to work correctly with my Work in progress changes from #2166.

So i assume it is related to the current lack of `#[cfg(xxx)]` handling.

`chrono-0.4.4/src/lib.rs`:
```rust
#[cfg(feature="clock")]
extern crate time as oldtime;

// this reexport is to aid the transition and should not be in the prelude!
pub use oldtime::Duration;

...
#[cfg(not(feature="clock"))]
mod oldtime;
```
`chrono-0.4.4/src/oldtime.rs`:
```rust
/// ISO 8601 time duration with nanosecond precision.
/// This also allows for the negative duration; see individual methods for details.
#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Debug)]
pub struct Duration {
    secs: i64,
    nanos: i32, // Always 0 <= nanos < NANOS_PER_SEC
}
```
`time-0.1.40/src/lib.rs`:
```rust
pub use duration::{Duration, OutOfRangeError};
...
mod duration;
```
`time-0.1.40/src/duration.rs`:
```rust
/// ISO 8601 time duration with nanosecond precision.
/// This also allows for the negative duration; see individual methods for details.
#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Debug)]
pub struct Duration {
    secs: i64,
    nanos: i32, // Always 0 <= nanos < NANOS_PER_SEC
}
```

The "clock" feature in chrono `Cargo.toml` and allows to choose between 2 implementations of `Duration`
```toml
[features]
clock = ["time"]
default = ["clock"]
```
With my branch, depending if the `clock` feature is enabled or disabled
`Duration` resolves to either `time-0.1.40/src/duration.rs` or to `chrono-0.4.4/src/oldtime.rs`

**Comment by ortem** [comment]
Seems to be fixed by #4635
